app/lib/features/chat/widgets/create_chat.dart

```diff
@@ -0,0 +1,684 @@
+      final config = sdk.newConvoSettingsBuilder();
+      final selectedUsers = ref.watch(_selectedUsersProvider);
+      // we check whether user has selected participants for DM/Group DM.
+      if (selectedUsers.isNotEmpty) {
+        if (selectedUsers.length > 1) {
+          // we are creating group DM
+          for (int i = 0; i < selectedUsers.length; i++) {
+            final userId = selectedUsers[i].userId().toString();
+            config.addInvitee(userId);
+          }
+          config.setName('');
+        } else {
+          // we are creating dm
+          final userId = selectedUsers[0].userId().toString();
+          config.setName('');
+          config.addInvitee(userId);
+        }
+      } else {
+        // we are creating default room
+        config.setName(convoName);
+      }
+      if (description.isNotEmpty) {
+        config.setTopic(description);
+      }
+      final avatarUri = ref.read(_avatarProvider);
+      if (avatarUri.isNotEmpty) {
+        config.setAvatarUri(avatarUri); // convo creation will upload it
+      }
+      final parentId = ref.read(selectedSpaceIdProvider);
+      if (parentId != null) {
+        config.setParent(parentId);
+      }
+      final client = ref.read(clientProvider);
+      if (client == null) throw UnimplementedError('Client is not available');
+      final roomId = await client.createConvo(config.build());
+      // add room to child of space (if given)
+      if (parentId != null) {
+        final space = await ref.read(spaceProvider(parentId).future);
+        await space.addChildSpace(roomId.toString());
+      }
+      final convo = await client.convoWithRetry(roomId.toString(), 12);
+      EasyLoading.dismiss();
+      EasyLoading.showSuccess(
+        'Chat Room Created with Room ID: ${convo.getRoomIdStr()}',
+      );
+      return convo;
+    } catch (e) {
+      EasyLoading.dismiss();
+      EasyLoading.showError(
+        'Some error occured $e',
+        duration: const Duration(seconds: 2),
+      );
+      return null;
+    }
+  }
+}
+
+///
+class _CreateChatWidget extends ConsumerWidget {
+  final PageController controller;
+  final Future<ffi.Convo?> Function(BuildContext, String, String) onCreateConvo;
+  const _CreateChatWidget({
+    required this.controller,
+    required this.onCreateConvo,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final selectedUsers = ref.watch(_selectedUsersProvider).toList();
+    final foundUsers = ref.watch(searchResultProvider);
+    final searchCtrl = ref.watch(searchController);
+    final largeWidth = isDesktop || MediaQuery.of(context).size.width > 600;
+    String tileTitle = selectedUsers.isEmpty
+        ? 'Start Chat'
+        : selectedUsers.length > 1
+            ? 'Start Group DM'
+            : checkUserDMExists(selectedUsers[0].userId().toString(), ref) !=
+                    null
+                ? 'Go in DM'
+                : 'Start DM';
```

Comment:
is there some way we could make that a little easier to understand? Maybe by having an explicit `makeTitle`-fn that has actually if-statements that are easier to follow than several nestings of `?` and `:` ?!?

Reply:
I will simplify it.